tests/scripts/generate_test_code.py

```diff
@@ -731,18 +732,19 @@ def gen_dep_check(dep_id, dep):
         raise GeneratorInputError("Dependency Id should be a positive "
                                   "integer.")
     _not, dep = ('!', dep[1:]) if dep[0] == '!' else ('', dep)
+    _defined = '' if re.search(r'(<=?|>=?|==)', dep) else 'defined'
```

Comment:
Rather than do a new search to detect an operator, I think the condition regex should define groups and you should access these groups.
```
CONDITION_OPERATOR_REGEX = r'[!=]=|[<>]=?'
CONDITION_VALUE_REGEX = r'[-+]?(0x[0-9A-Fa-f]+|0|[1-9][0-9]*)' # forbid 0ddd which might be accidentally octal or accidentally decimal
CONDITION_REGEX = r'({})(?:\s*({})\s*({}))?'.format(C_IDENTIFIER_REGEX, CONDITION_OPERATOR_REGEX, CONDITION_VALUE_REGEX)

m = re.match(CONDITION_REGEX, …)
if not m: …
_defined = '' if m.group(2) else 'defined'
if m.group(2):
    value = int(m.group(3), 0)
```

Reply:
Thanks! I'll do that

Reply:
The problem with this pattern, is that invalid strings, such as `ABC%8192` also pass the validate test

Reply:
Nonetheless, there is a need for a new search. The first check is to determine the `_not` and then we need to find if `dep` is  a condition or a definition

Reply:
need to add '$' at the end of the REGEX